### PR TITLE
Addressing fix for AspNetCoreResponseCookies cookie defaults

### PR DESCRIPTION
**Pull request by surgupta-msft**
### Issue describing the changes in this PR

resolves #[issue_for_this_pr](https://github.com/orgs/Azure/projects/473/views/9?sliceBy%5Bvalue%5D=surgupta-msft&pane=issue&itemId=80813262)

Addressing issue - https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca5383

Guidance for the changes is shared in Codeql links from Engineering hub - [Link1](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/codeql/codeql-semmle), [Link2](https://eng.ms/docs/cloud-ai-platform/devdiv/serverless-paas-balam/serverless-paas-vikr/app-service-web-apps/app-service-team-documents/generalteamdocs/security/sdl/codeql)

Once the PR and justification look good, will send it to Anirudh for approval as per the guidance [here](https://strikecommunity.azurewebsites.net/articles/9993/continuous-sdl-cai-security-exceptions.html)

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by jviau — APPROVED** [review]

File: extensions/Worker.Extensions.Http.AspNetCore/release_notes.md [hunk 1]
```diff
@@ -6,8 +6,7 @@
 
 ### Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore <version>
 
-- Updated`Updated Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore.Analyzers` 1.0.3
+- Addressing fix for AspNetCoreResponseCookies cookie defaults (#2811)
 
-### Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore.Analyzers 1.0.3
+### Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore.Analyzers  <version>
```
Comment (satvu) [inline extensions/Worker.Extensions.Http.AspNetCore/release_notes.md:11]: Discussed offline - this change is not yet released so it should still be included in the release notes.

Comment (fabiocav) [reply extensions/Worker.Extensions.Http.AspNetCore/release_notes.md:11]: So, to confirm, this wasn't released with 1.0.3? https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore.Analyzers/1.0.3

**Review by satvu — APPROVED** [review]

**Review by liliankasem — APPROVED** [review]

**Comment by anirudhgarg** [comment]
Approved as this is just forwarding an existing cookie and we can set up something on behalf of the customer.

**Review by anirudhgarg — APPROVED** [review]
Approved CodeQL exception as we don't want to change the behavior of an existing cookie.

**Pull request merged**